### PR TITLE
Add backend for quests

### LINKED ISSUE #141: Add quest backend
Add a way for a quest to be given to a user through a qr code in their profile.

### PR DESCRIPTION
**Pull request by ttoino**
Closes #141.

### CODE REVIEW

**Review by Naapperas — DISMISSED** [review]

File: app/Http/Controllers/EventController.php [hunk 1]
```diff
@@ -39,7 +42,7 @@ public function join(Request $request, Event $event)
 
         // FIXME: when attempting to join multiple times in a row, only the first attempt triggers the banner.
         if ($user->cannot('join', $event)) {
-            return redirect()->back()->dangerBanner('Não pode inscrever-se neste evento.');
+            return redirect()->back()->dangerBanner('Não podes inscrever-se neste evento');
```
Comment (Naapperas) [inline app/Http/Controllers/EventController.php]:
```suggestion
            return redirect()->back()->dangerBanner('Não podes inscrever-te neste evento');
```

File: app/Http/Controllers/QuestController.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+        $edition = $request->input('edition');
+        $participant = Participant::firstWhere('quest_code', $request->get('quest_code'));
+        $enrollment = $participant->enrollments()->where('edition_id', $edition?->id)->first();
+
+        Gate::authorize('give', [$quest, $enrollment]);
+
+        $enrollment->quests()->attach($quest);
+
+        return redirect()->back()->banner('Quest atribuída com sucesso!');
```
Comment (Naapperas) [inline app/Http/Controllers/QuestController.php:26]: Beware of text translations.

File: app/Providers/AuthServiceProvider.php [hunk 3]
```diff
@@ -45,5 +49,24 @@ public function boot(): void
                 $event->enrollments()->count() < $event->capacity // event must not be full
             )
         ));
+
+        Gate::define('give', fn (User $user, Quest $quest, Enrollment $enrollment) => (
+            $enrollment->quests()->where('quest_id', $quest->id)->doesntExist() && // participant must not have the quest
+            ( // either
+                (
+                    $user->isAdmin() && // user is admin and either
```
Comment (Naapperas) [inline app/Providers/AuthServiceProvider.php:57]: Shouldn't this allow for staff to perform this action?

**Review by Naapperas — CHANGES_REQUESTED** [review]

File: app/Providers/AuthServiceProvider.php (hunk 3 above)
Comment (ttoino) [reply app/Providers/AuthServiceProvider.php:57]: Yes, but that's a little hard right now, we should do that in a bigger PR so staff have all the necessary permissions.

File: app/Http/Controllers/QuestController.php (hunk 2 above)
Comment (ttoino) [reply app/Http/Controllers/QuestController.php:26]: We're not translating any banners right now, maybe #75 can handle that

File: app/Http/Controllers/EventController.php (hunk 1 above)
Comment (ttoino) [reply app/Http/Controllers/EventController.php]: You're right, mb

**Review by Naapperas — APPROVED** [review]

**Pull request merged**